core/src/main/java/de/danoeh/antennapod/core/service/FeedMediaSizeService.java

```diff
@@ -15,50 +31,59 @@ public FeedMediaSizeService() {
-//                    e.printStackTrace();
-//                } finally {
-//                    if (conn != null) {
-//                        conn.disconnect();
-//                    }
-//                }
-//            }
-//            if (size <= 0) {
-//                // they didn't tell us the size, but we don't want to keep querying on it
-//                media.setCheckedOnSizeButUnknown();
-//            } else {
-//                media.setSize(size);
-//            }
-//            Log.d(TAG, "Size now: " + media.getSize());
-//            DBWriter.setFeedMedia(this, media);
-//            EventBus.getDefault().post(FeedMediaEvent.update(media));
-//        }
+        if(false == NetworkUtils.isDownloadAllowed()) {
+            return;
+        }
+        List<FeedMedia> list = DBReader.getFeedMediaUnknownSize(this);
+        for (FeedMedia media : list) {
+            Log.d(TAG, "Getting size currently " + media.getSize() + " for " + media.getDownload_url());
+            if(false == NetworkUtils.isDownloadAllowed()) {
+                return;
+            }
+            long size = Integer.MIN_VALUE;
+            if (media.isDownloaded()) {
+                File mediaFile = new File(media.getLocalMediaUrl());
+                if(mediaFile.exists()) {
+                    size = mediaFile.length();
+                }
+            } else if (false == media.checkedOnSizeButUnknown()) {
+                // only query the network if we haven't already checked
+                OkHttpClient client = AntennapodHttpClient.getHttpClient();
+                Request.Builder httpReq = new Request.Builder()
+                        .url(media.getDownload_url())
+                        .header("Accept-Encoding", "identity")
+                        .head();
+                try {
+                    Response response = client.newCall(httpReq.build()).execute();
+                    if(response.isSuccessful()) {
+                        String contentLength = response.header("Content-Length");
+                        try {
+                            size = Integer.parseInt(contentLength);
+                        } catch(NumberFormatException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+            if (size <= 0) {
+                // they didn't tell us the size, but we don't want to keep querying on it
+                media.setCheckedOnSizeButUnknown();
+            } else {
+                media.setSize(size);
+            }
+            Log.d(TAG, "Size now: " + media.getSize());
+            DBWriter.setFeedMedia(this, media);
+            EventBus.getDefault().post(FeedMediaEvent.update(media));
+
+            // we don't want to stress the server too much
+            try {
+                Thread.sleep(1500);
```

Comment:
Won't this lead to battery drain?  Instead of the usual recommend approach of doing all our work at once we're spreading it out over a long time, preventing the device from sleeping.